File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int colonIndex = logLine.IndexOf(':');
        return logLine.Substring(colonIndex + 1).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int startIndex = logLine.IndexOf('[') + 1;
        int endIndex = logLine.IndexOf(']');
        return logLine.Substring(startIndex, endIndex - startIndex).ToLower();
    }

    public static string Reformat(string logLine)
    {
        string message = Message(logLine);
        string level = LogLevel(logLine);
        return $"{message} ({level})";
    }
}

File: solutions/cpp/log-levels/2/log_levels.cpp
#include <iostream>
#include <string>

namespace log_line {

const std::string delimiter = ": ";

std::string message(std::string line) {
  // return the message
  return line.substr(line.find(delimiter) + delimiter.size());
}

std::string log_level(std::string line) {
  // return the log level
  return line.substr(1, line.find(delimiter) - delimiter.size());
}

std::string reformat(std::string line) {
  // return the reformatted message
  return message(line) + " (" + log_level(line) + ")";
}
} // namespace log_line
